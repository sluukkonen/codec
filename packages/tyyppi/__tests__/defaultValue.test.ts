import { defaultValue, string } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

test("should return the default if decoding undefined", () => {
  expectParseSuccess(defaultValue(string, "abc"), undefined, "abc")
})

test("should defer to the underlying codec if the value is not undefined", () => {
  expectParseSuccess(defaultValue(string, "abc"), "foo")
  expectParseFailure(defaultValue(string, "abc"), null)
})
