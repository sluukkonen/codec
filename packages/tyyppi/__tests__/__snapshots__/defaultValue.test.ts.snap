// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should defer to the underlying codec if the value is not undefined 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "string",
    "message": "Expected a string, received null",
    "path": [],
    "received": "null",
  },
]
`;
